GitUI/UserControls/FileStatusList.cs

```diff
@@ -1117,25 +1117,43 @@ private void FileStatusListView_DrawSubItem(object sender, DrawListViewSubItemEv
 
             if (!string.IsNullOrEmpty(text))
             {
-                var slashIndex = text.LastIndexOf('/');
-
-                var textRect = new Rectangle(textStartX, item.Bounds.Top, textMaxWidth, item.Bounds.Height);
+                string suffix = null;
+                {
+                    int parenIndex = text.IndexOf(" ("); // renamed/copied file
+                    if (parenIndex >= 0 && parenIndex < text.Length)
+                    {
+                        suffix = text.Substring(parenIndex);
+                        text = text.Substring(0, parenIndex);
+                    }
+                }
 
-                if (slashIndex == -1 || slashIndex >= text.Length - 1)
+                string prefix = null;
                 {
-                    formatter.DrawString(text, textRect, SystemColors.ControlText);
-                    return;
+                    int slashIndex = text.LastIndexOf('/');
+                    if (slashIndex >= 0 && slashIndex < text.Length)
+                    {
+                        prefix = text.Substring(0, slashIndex + 1);
+                        text = text.Substring(slashIndex + 1);
+                    }
```

Comment:
Let's extract prefix/suffix calculations into own private method and test these extensively (via `TestAccessor`)

Reply:
How to extract the old name safely? Let PathFormatter return it separately! Rewritten including test for new methods.